### PR TITLE
improvement: Quiz resize animations

### PR DESCRIPTION
**Pull request by mathmods**
**Please check if the PR fulfills these requirements:**

- [x] The commit message follows our guidelines.
- For bug fixes and features:
    - [x] You tested the changes.
    - [ ] You updated the docs or changelog.

Related issue (if applicable): #15 

**What kind of change does this PR introduce?**
Animates resizes of Quiz content blocks. Reacts upon any change of choice view or result view and will proceed to fit (if they are visible).

**Does this PR introduce a breaking change?**
No

## New feature or change ##

**What is the current behavior?** 
Instantaneous jumps to new size.

**What is the new behavior?**
Animated size changes of rectangle size to fit the current view.

**Other information**
Letters sometimes gets a black line over them for no apparent reason. Not sure if this is created on my own pc running this or if this is introduced by BoundaryControl. Affects text on quiz answer elements and quiz label.

You can see the quiz unfolding itself if you scroll fast enough. (I couldn't find a way to set the correct size without waiting for the views to update with the loaded content.)

Some answers with long text might wrap around once selected and increase the size of the view creating a slightly weird looking reaction by the container. (Not breaking anything, looks normal after .5 seconds)

Will work with hints when implemented.

### CODE REVIEW

File: ui/screens/lesson/UIBaseQuiz.gd [hunk 1]
```diff
@@ -8,37 +8,47 @@ const ERROR_OUTLINE := preload("res://ui/theme/quiz_outline_error.tres")
...
+onready var _error_tween := $ErrorTween as Tween
+onready var _size_tween := $SizeTween as Tween
+onready var _help_message := $MarginContainer/BoundaryControl/ChoiceView/HelpMessage as Label
 
 var _quiz: Quiz
 var _shake_pos: float = 0
+# For animating changing the size of the container
+var _previous_rect_size := rect_size
+var _next_rect_size := Vector2.ZERO
+var _percent_transformed: float = 0
```
Comment (NathanLovato) [inline ui/screens/lesson/UIBaseQuiz.gd:51]: You can use := 0.0 to get float type inference

**Comment by mathmods** [comment]
Hmm. It seems to only be present if the content is set to something. I got it to print the sizes of everything in the tree and it practically told me that nothing was using that space. Essentially, choice_view isn't updating it's own size to match? Or rather it just keeps it size? I'll mess about with this to see if it leads me somewhere

**Comment by mathmods** [comment]
Ok. Got the gap to become smaller by forcing a resize, but it's still there. Approximately 1 line instead of 2~3.
Removing the expand flag from vertical makes the gap appear at the bottom and not in the middle which is confusing me to no end.

I'm considering just cheating by removing the "expand" flag on the rich label and then assume a smaller size if content is visible.

**Comment by mathmods** [comment]
Ok, I noticed this is affecting different lessons in different ways. With the newest commit: any content with 1 line will display an additional line just beneath it. Any content larger than that and it's normal. 

In other words; I see 3 possible solutions:

* 1 line once in a while isn't that bad, just commit this as it is.
* Try and add an empty line before the content so that it takes some distance from the header, instead of the answers having distance to the content.
* Try and detect 1 line contents and "cheat" as I mentioned in the previous comment.

**Comment by mathmods** [comment]
Ok, I have found a better solution that completely removes it.
Sadly it will change how the animation works since it moves the boundary to become the parent of the margin container and not vice versa.

It will take some time, but I'm going to make it work properly.

**Comment by NathanLovato** [comment]
Thanks for taking the time, we truly appreciate it. I'll be there to review and merge your new PR when it's ready.

**Pull request closed without merging**